prow/repoowners/repoowners.go

```diff
@@ -241,10 +261,20 @@ func (c *Client) LoadRepoOwners(org, repo, base string) (RepoOwner, error) {
 		}
 		entry.owners, err = loadOwnersFrom(gitRepo.Dir, mdYaml, entry.aliases, dirBlacklist, log)
 		if err != nil {
-			return nil, fmt.Errorf("failed to load RepoOwners for %s: %v", fullName, err)
+			return fmt.Errorf("failed to load RepoOwners for %s: %v", fullName, err)
 		}
 		entry.sha = sha
 		c.cache[fullName] = entry
+		return nil
+	}
+
+	if !ok || entry.sha == "" || entry.owners == nil || entry.owners.enableMDYAML != mdYaml || entry.aliases == nil {
+		// load entry without doing git diff
+		syncEntry(false)
```

Comment:
Instead of passing in a `bool` and doing conditional work in the delegate, can we just do the work in the reload case upfront and pass the data to `syncEntry`?

Reply:
fixed